ecl/eclscheduler/eclscheduler.cpp

```diff
@@ -180,6 +181,258 @@ version: "1.0"
+                    }
+
+                    if (newQueues)
+                        updateSchedulerQueues();
+                }
+                catch (IException *E)
+                {
+                    EXCLOG(E);
+                    releaseAtoms();
+                    ExitModuleObjects();
+                    _exit(2);
+                }
+                catch (...)
+                {
+                    IERRLOG("Unknown exception caught in eclscheduler processQueues - restarting");
+                    releaseAtoms();
+                    ExitModuleObjects();
+                    _exit(2);
+                }
+            }
+        }
+        DBGLOG("eclscheduler closing");
+        stop();
+    }
+
+private:
+    StringBuffer &getQueues(const IPropertyTree *config, StringBuffer &queueNameList)
+    {
+#ifdef _CONTAINERIZED
+        Owned<IPTreeIterator> queues = config->getElements("queues");
+        ForEach(*queues)
+        {
+            IPTree &queue = queues->query();
+            const char *qname = queue.queryProp("@name");
+            if (queueNameList.length())
+                queueNameList.append(",");
+            queueNameList.append(qname);
+        }
+#else
+        const char *processName = config->queryProp("@name");
+        queueNameList.append(processName);
+#endif
+        return queueNameList;
+    }
+
+    void updateSchedulerQueues()
+    {
+        std::unordered_set<std::string> updatedQueuesSet = splitBy(currentQueueNames.str(), ",");
+
+        // stop and remove schedulers not in the new list as they are no longer needed
+        // for those in the new list, if they are already running, leave them alone
+        std::list<aindex_t> schedulersToBeRemoved;
+        ForEachItemIn(schedIdx, schedulers)
+        {
+            if (updatedQueuesSet.find(schedulers.item(schedIdx).getServerName().str()) == updatedQueuesSet.end())
+            {
+                // Queue has been removed
+                DBGLOG("Stopped listening to queue %s", schedulers.item(schedIdx).getServerName().str());
+                schedulers.item(schedIdx).stop();
+                schedulersToBeRemoved.emplace_front(schedIdx);
+            }
+            else
+            {
+                // "New" queue already running
+                updatedQueuesSet.erase(schedulers.item(schedIdx).getServerName().str());
+            }
+        }
+        for (auto &index : schedulersToBeRemoved)
+        {
+            schedulers.remove(index);
+        }
+
+        // only new schedulers will be left in the list
+        for (auto &qname : updatedQueuesSet)
+        {
+#ifdef _CONTAINERIZED
+            DBGLOG("Start listening to queue %s", qname.c_str());
+            Owned<EclScheduler> scheduler = new EclScheduler(this, qname.c_str());
+            scheduler->start();
+            schedulers.append(*scheduler.getClear());
+#else
+            Owned<IStringIterator> targetClusters = getTargetClusters("EclSchedulerProcess", qname.c_str());
```

Comment:
something wrong here.. the BM queue names, should have been fetched already, this code should look the same for BM and containered, walking the new queues list and creating EclScheduler's based on the names in the queue.